### DISCUSSION TITLE
Problem with Harlem and Nuxt 3

### DISCUSSION THREAD

**Discussion by Zaklo** [andrewcourtice/harlem#46] [Q&A] [answered]
Hey ! 

I'm experiencing some troubles with Harlem and Nuxt 3.

Both works perfectly when I `npm run dev`, but when I `build` and then `start`, I have this issue : 

`Named export 'createStore' not found. The requested module '@harlem/core' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:`

`import pkg from '@harlem/core';
const { createStore } = pkg;`

` at ModuleJob._instantiate (internal/modules/esm/module_job.js:121:21)  
  at async ModuleJob.run (internal/modules/esm/module_job.js:166:5)  
  at async Loader.import (internal/modules/esm/loader.js:178:24)  
  at async file://./.output/server/chunks/app/render.mjs:527:24  
  at async renderMiddleware (file://./.output/server/chunks/app/render.mjs:576:20)  
  at async handle (file://./.output/server/node_modules/h3/dist/index.mjs:601:19)`

I'm not sure to understand why this happen.

I'm importing createStore -> `import {createStore} from '@harlem/core'`

If anyone has an idea, I'll be happy to ear :) 

Thanks !

**Comment by Zaklo** [comment, marked as answer]
@andrewcourtice any clue ?

**Reply by andrewcourtice** [reply]
Hi @Zaklo 

Apologies for the delayed response.

I'm honestly not sure what may be happening in your case. It looks as though it's a build issue. Do you know if your Nuxt environment is configured to build with Webpack or with Vite?

If your Nuxt env is configured to use Vite, try adding this to your `nuxt.config.ts` and let me know if that works:
```typescript
export default defineNuxtConfig({
  vite: {
    build: {
      commonjsOptions: {
        esmExternals: true,
      },
    },
  },
});
```

Thanks

**Reply by Zaklo** [reply]
Hey @andrewcourtice !

Thanks for the reply :) I resolved my problem, I had to transpile harlem in nuxt config : 

`build: {
        transpile: ['@harlem/core', '@harlem/extension-action']
    }`

So now it works perfectly :)

**Reply by andrewcourtice** [reply]
No worries @Zaklo 😄  Glad you got it sorted. If you need help with anything else just let me know 👍 Hope you enjoy using Harlem 😄

**Reply by andrewcourtice** [reply]
@Zaklo I've just been made aware that the Nuxt community team are working on a module for Harlem. This should simplify the process of integrating Harlem with your Nuxt project.

The module is available here: https://github.com/nuxt-community/harlem-module

**Reply by Zaklo** [reply]
Oh nice ! thanks @andrewcourtice !! :)
